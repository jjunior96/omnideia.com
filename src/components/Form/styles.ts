import styled, { css } from 'styled-components';

import * as ButtonComponent from 'components/Button/styles';
import * as InputComponent from 'components/Input/styles';

export const Container = styled.form``;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    flex: 1;

    width: 100%;

    padding: 0 ${theme.spacings.small};

    ${InputComponent.Container} {
      margin-bottom: ${theme.spacings.xxsmall};
    }

    ${ButtonComponent.Container} {
      margin-top: ${theme.spacings.xxsmall};
    }

    label {
      margin-top: ${theme.spacings.xsmall};
    }

    button[type='submit'] {
      margin: ${theme.spacings.small} 0;
    }
  `}
`;
