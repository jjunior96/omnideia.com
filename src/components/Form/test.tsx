import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Form from '.';

describe('<Form />', () => {
  it('should render the Form', () => {
    const { container } = renderWithTheme(
      <Form aria-label="User info" name="userinfo">
        <input aria-label="username" placeholder="Username" />
      </Form>
    );

    expect(
      screen.getByRole('textbox', { name: /username/i }).parentElement
    ).toHaveStyle({
      display: 'flex',
      'flex-direction': 'column',

      width: '100%',

      padding: '0 2.4rem'
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
