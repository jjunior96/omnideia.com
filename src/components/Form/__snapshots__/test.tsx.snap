// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render the Form 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  padding: 0 2.4rem;
}

.c0 label {
  margin-top: 1.6rem;
}

.c0 button[type='submit'] {
  margin: 2.4rem 0;
}

<form
  class=""
>
  <div
    class="c0"
  >
    <input
      aria-label="username"
      placeholder="Username"
    />
  </div>
</form>
`;
